### ISSUE TITLE
Low Test/Accuracy Value (Avocado Ripeness)

### ISSUE THREAD

**Issue by NunoFernandesEEC** [cogsys-tuebingen/deephs_fruit#17]
Hello, my name is Nuno, and I am a bachelor’s student. Your article, "Measuring the Ripeness of Fruit with Hyperspectral Imaging and Deep Learning," closely relates to my final project, although my goal is to classify skin lesions. Since I don’t yet have a dataset, I am attempting to replicate your work.

I have trained and tested your Deephs_net algorithm, and I obtained results similar to those reported by others, but not identical to yours, for avocado ripeness. Specifically, my results were: training accuracy ~92%, validation accuracy ~86%, and test accuracy ~26.5%. Upon analyzing the results, I noticed that the dataset appears to be imbalanced, with a significantly larger number of images in the unripe class compared to the others in training dataset. 

Would you mind sharing which parameters you changed in your code on GitHub? Alternatively, if you could provide the weights, I would greatly appreciate it.

If there is any additional information you need, please feel free to let me know.

The image below shows the results I obtained using the same parameters as those in your paper, "Measuring the Ripeness of Fruit with Hyperspectral Imaging and Deep Learning," except for the first Conv2D layer in Fig. 4, where I used a kernel size of 7 instead of 3, as you did in the paper. All other parameters are the same.

![Image](https://github.com/user-attachments/assets/b3a5c42a-f764-4f46-812c-d4e64e1f0d84)

**Comment by leonvarga** [comment]
Hi NunoFernandesEEC,

Your observation is correct. There are far more unripe recordings than ripe ones.
This is why we oversample the other classes. This is done by the HyperspectralDataset (https://github.com/cogsys-tuebingen/deephs_fruit/blob/master/core/hyperspectral_dataset.py).
See also https://github.com/cogsys-tuebingen/deephs_fruit/blob/5f47812a3fb5bd8f0ed5367cedf9b464f83076c1/classification/train.py#L74.

If you do not balance all the classes during training, a bias is introduced for the unripe class, which could explain your results.

I therefore assume that you have your own data loader pipeline running. If you add the oversampling logic, it should solve the issue.

Best regards,
Leon

**Comment by leonvarga** [comment]
You can find some pretrained weights and more models here: https://github.com/cogsys-tuebingen/hsi_benchmark
